runtime/lib/ttmetal/command_queue.cpp

```diff
@@ -141,19 +141,85 @@ static void writeFile(std::string const &fileName, char const *data,
 
 static std::variant<DataMovementConfig, ComputeConfig, EthernetConfig>
 createKernelConfig(::tt::target::metal::KernelSource const *kernelSource) {
-  switch (kernelSource->source_type()) {
-  case ::tt::target::metal::SourceType::Noc0: {
-    return ::tt::tt_metal::ReaderDataMovementConfig();
+  switch (kernelSource->config_type()) {
+  case ::tt::target::metal::KernelConfig::NocConfig: {
+    switch (kernelSource->config_as_NocConfig()->noc_index()) {
+    case (tt::target::metal::NocIndex::Noc0): {
+      ::tt::tt_metal::ReaderDataMovementConfig readerDataMovementConfig;
+      readerDataMovementConfig.noc = NOC::NOC_0;
+      return readerDataMovementConfig;
+    }
+    case (tt::target::metal::NocIndex::Noc1): {
+      ::tt::tt_metal::WriterDataMovementConfig writerDataMovementConfig;
+      writerDataMovementConfig.noc = NOC::NOC_1;
+      return writerDataMovementConfig;
+    }
+      throw std::runtime_error("Unsupported noc index");
+    }
   }
-  case ::tt::target::metal::SourceType::Noc1: {
-    return ::tt::tt_metal::WriterDataMovementConfig();
+  case ::tt::target::metal::KernelConfig::EthernetConfig: {
+    ::tt::tt_metal::EthernetConfig ethernetConfig;
+    switch (kernelSource->config_as_EthernetConfig()->eth_type()) {
+    case (tt::target::metal::EthType::Sender): {
+      ethernetConfig.eth_mode = Eth::SENDER;
+      break;
+    }
+    case (tt::target::metal::EthType::Receiver): {
+      ethernetConfig.eth_mode = Eth::RECEIVER;
+      break;
+    }
+      throw std::runtime_error("Unsupported eth type");
+    }
+
+    switch (kernelSource->config_as_EthernetConfig()->noc_index()) {
+    case (tt::target::metal::NocIndex::Noc0): {
+      ethernetConfig.noc = NOC::NOC_0;
+      break;
+    }
+    case (tt::target::metal::NocIndex::Noc1): {
+      ethernetConfig.noc = NOC::NOC_1;
+      break;
+    }
+      throw std::runtime_error("Unsupported noc index");
+    }
+    return ethernetConfig;
   }
-  case ::tt::target::metal::SourceType::Tensix: {
-    return ::tt::tt_metal::ComputeConfig();
+
+  case ::tt::target::metal::KernelConfig::TensixConfig: {
+    ::tt::tt_metal::ComputeConfig computeConfig;
```

Comment:
I wonder if we should do this to get zero-initialized fields in this struct and others?

```
::tt::tt_metal::ComputeConfig computeConfig = {};
```

My worry is how to detect if more members are added to `ComputeConfig` struct in the future, the best way would be to use it's constructor I think (this would require moving the switch statement on individual fields like math_fidelity out of here, so they can be set with a single function call though.  @nsmithtt what do you think about structs being initialized the way they are here?